substreams/rpccalls.go

```diff
@@ -34,19 +34,26 @@ func (e *RPCExtensioner) Params() map[string]string {
 }
 
 func (e *RPCExtensioner) WASMExtensions(in map[string]string) (map[string]map[string]wasm.WASMExtension, error) {
-	// set default values from flags ????
-	switch len(in) {
-	case 0:
-		return nil, nil
-	case 1:
-		break
-	default:
-		return nil, fmt.Errorf("unsupported wasm extensions: %v (only 'rpc_eth_call' is implemented)", in)
+	// Accept either or both rpc_eth_call and rpc_eth_get_balance, must match if both provided
+	rpcInfoCall, okCall := in["rpc_eth_call"]
+	rpcInfoBal, okBal := in["rpc_eth_get_balance"]
+	if !okCall && !okBal {
+		return nil, fmt.Errorf("unsupported wasm extensions: %v (only 'rpc_eth_call' & 'rpc_eth_get_balance' are implemented)", in)
 	}
 
-	rpcInfo, found := in["rpc_eth_call"]
-	if !found {
-		return nil, fmt.Errorf("unsupported wasm extensions: %v (only 'rpc_eth_call' is implemented)", in)
+	var rpcInfo string
+	if okCall && okBal {
+		if rpcInfoCall != rpcInfoBal {
```

Comment:
No no no! If you support distinct flags, you will need to support them all the way ! NewRPCEngine  should accept a list of URLs for Call and another list for Balance, supporting one or the other if either list is nil
so you keep both rpcInfo, split them with a function that returns an array, pass these arrays to NewRPCEngine...